doc/classes/String.xml

```diff
@@ -344,111 +385,126 @@
+				Returns [code]true[/code] if this string is a valid hexadecimal number. A valid hexadecimal number only contains digits or letters [code]A[/code] to [code]F[/code] (either uppercase or lowercase), and may be prefixed with a positive ([code]+[/code]) or negative ([code]-[/code]) sign.
+				If [param with_prefix] is [code]true[/code], the hexadecimal number needs to prefixed by [code]"0x"[/code] to be valid.
+				[codeblock]
+				print("A08E".is_valid_hex_number())    # Prints true
+				print("-AbCdEf".is_valid_hex_number()) # Prints true
+				print("2.5".is_valid_hex_number())     # Prints false
+
+				print("0xDEADC0DE".is_valid_hex_number(true)) # Prints true
+				[/codeblock]
 			</description>
 		</method>
 		<method name="is_valid_html_color" qualifiers="const">
 			<return type="bool" />
 			<description>
-				Returns [code]true[/code] if this string contains a valid color in hexadecimal HTML notation. Other HTML notations such as named colors or [code]hsl()[/code] colors aren't considered valid by this method and will return [code]false[/code].
+				Returns [code]true[/code] if this string is a valid color in hexadecimal HTML notation. The string must be a hexadecimal value (see [method is_valid_hex_number]) of either 3, 4, 6 or 8 digits, and may be prefixed by a hash sign ([code]#[/code]). Other HTML notations for colors, such as names or [code]hsl()[/code], are not considered valid. See also [method Color.html].
 			</description>
 		</method>
 		<method name="is_valid_identifier" qualifiers="const">
 			<return type="bool" />
 			<description>
-				Returns [code]true[/code] if this string is a valid identifier. A valid identifier may contain only letters, digits and underscores ([code]_[/code]) and the first character may not be a digit.
+				Returns [code]true[/code] if this string is a valid identifier. A valid identifier may contain only letters, digits and underscores ([code]_[/code]), and the first character may not be a digit.
 				[codeblock]
-				print("good_ident_1".is_valid_identifier()) # Prints "true"
-				print("1st_bad_ident".is_valid_identifier()) # Prints "false"
-				print("bad_ident_#2".is_valid_identifier()) # Prints "false"
+				print("node_2d".is_valid_identifier())    # Prints true
+				print("TYPE_FLOAT".is_valid_identifier()) # Prints true
+				print("1st_method".is_valid_identifier()) # Prints false
+				print("MyMethod#2".is_valid_identifier()) # Prints false
 				[/codeblock]
 			</description>
 		</method>
 		<method name="is_valid_int" qualifiers="const">
 			<return type="bool" />
 			<description>
-				Returns [code]true[/code] if this string contains a valid integer.
+				Returns [code]true[/code] if this string represents a valid [int]. A valid integer only contains digits, and may be prefixed with a positive ([code]+[/code]) or negative ([code]-[/code]) sign. See also [method to_int].
```

Comment:
Unsure. The GDScript compiler is one thing and the **int** type is another. Many different languages may define their own special way to write an integer, but at its core, in Godot an **int** is an integer and a **float** is a floating-point value.

Reply:
Yeah I was GDScript-specific here - anyway, since this method has its own rules, I'd say it's better to just state those.

Reply:
Come to think of it, one method was renamed from 3.x. It used to be called `is_valid_integer`. The current name is a bit of a misnomer because it could really just go above the 64-bit limit if someone wanted to use it for that purpose... So it's not just about the Variant Type called `int`, it's any kind of integer.